src/allotropy/parsers/moldev_softmax_pro/softmax_pro_parser.py

```diff
@@ -1,17 +1,266 @@
+                ],
+                calculated_data_aggregate_document=None,
+            ),
+        )
+
+    def _get_plate_reader_document_item(
+        self, plate_block: PlateBlock, position: str
+    ) -> PlateReaderDocumentItem:
+        if plate_block.plate_block_type == "Absorbance":
+            return PlateReaderDocumentItem(
+                measurement_aggregate_document=MeasurementAggregateDocument(
+                    measurement_time=EPOCH,
+                    plate_well_count=TQuantityValueNumber(plate_block.header.num_wells),
+                    container_type=ContainerType.well_plate,
+                    measurement_document=list(
+                        self._iter_absorbance_measurement_document(
+                            plate_block, position
+                        ),
+                    ),
+                )
+            )
+        elif plate_block.plate_block_type == "Luminescence":
+            return PlateReaderDocumentItem(
+                measurement_aggregate_document=MeasurementAggregateDocument(
+                    measurement_time=EPOCH,
+                    plate_well_count=TQuantityValueNumber(plate_block.header.num_wells),
+                    container_type=ContainerType.well_plate,
+                    measurement_document=list(
+                        self._iter_luminescence_measurement_document(
+                            plate_block, position
+                        ),
+                    ),
+                )
+            )
+        elif plate_block.plate_block_type == "Fluorescence":
+            return PlateReaderDocumentItem(
+                measurement_aggregate_document=MeasurementAggregateDocument(
+                    measurement_time=EPOCH,
+                    plate_well_count=TQuantityValueNumber(plate_block.header.num_wells),
+                    container_type=ContainerType.well_plate,
+                    measurement_document=list(
+                        self._iter_fluorescence_measurement_document(
+                            plate_block, position
+                        )
+                    ),
+                )
+            )
+        else:
+            error = "Unable to find valid plate block type."
+            raise AllotropeConversionError(error)
+
+    def _iter_fluorescence_measurement_document(
+        self, plate_block: PlateBlock, position: str
+    ) -> Iterator[FluorescencePointDetectionMeasurementDocumentItems]:
+        if plate_block.header.scan_position == "TRUE":
+            scan_position = (
+                ScanPositionSettingPlateReader.bottom_scan_position__plate_reader_
+            )
+        elif plate_block.header.scan_position == "FALSE":
+            scan_position = (
+                ScanPositionSettingPlateReader.top_scan_position__plate_reader_
+            )
+        else:
+            error = "Unable to find valid scan position."
+            raise AllotropeConversionError(error)
```

Comment:
Could you extract this block to a helper -- e.g., `_get_fluorescence_plate_block_scan_position`?